src/main/java/io/gravitee/common/utils/RxHelper.java

```diff
@@ -37,4 +43,67 @@ public class RxHelper {
     public static <R> FlowableTransformer<R, R> mergeWithFirst(Flowable<R> other) {
         return upstream -> other.materialize().mergeWith(upstream.materialize()).dematerialize(n -> n);
     }
+
+    /**
+     * Returns a {@link FlowableTransformer} that can be used in a composition.
+     * It applies an interval for each element received in the {@link Flowable} upstream in opposition to {@link Flowable#delay(long, TimeUnit)} which will apply the delay only once for the {@link Flowable}
+     *
+     * @param delay the delay to apply between each element of the {@link Flowable}
+     * @param timeUnit the {@link TimeUnit}  of the delay
+     *
+     * @return a {@link FlowableTransformer} that will be applied.
+     */
+    public static <R> FlowableTransformer<R, R> delayElement(int delay, TimeUnit timeUnit) {
+        return upstream -> upstream.zipWith(Flowable.interval(delay, timeUnit), (item, interval) -> item);
+    }
+
+    /**
+     * Returns a {@link FlowableTransformer} that can be used in a composition.
+     * It retries the {@link Flowable} X times with delay between each attempt
+     *
+     * @param times the attempts number
+     * @param backOffDelay the delay between each retry
+     * @param timeUnit the {@link TimeUnit} of the backOffDelay
+     *
+     * @return a {@link FlowableTransformer} that will be applied.
+     */
+    public static <R> FlowableTransformer<R, R> retryFlowable(int times, int backOffDelay, TimeUnit timeUnit) {
```

Comment:
You named the args "backOffDelay" but I don't see any backoff strategy here, only the same delay applies again and again. Shouldn't we implement a proper exponentiel backoff strategy with min/max delay and jitter multiplier?

Reply:
backOff strategies will be implemented with the retry feature as discussed.
the variable will be named `retryInterval` to avoid confusion